activation/activation.go

```diff
@@ -394,47 +443,82 @@ func (b *Builder) PublishActivationTx(ctx context.Context) error {
 		b.log.With().Info("building new atx challenge", b.currentEpoch())
 		err := b.buildNipstChallenge(ctx)
 		if err != nil {
-			b.log.With().Error("failed to build new atx challenge", log.Err(err))
+			return fmt.Errorf("failed to build new atx challenge: %w", err)
+		}
+	}
+
+	if b.pendingATX == nil {
+		var err error
+		b.pendingATX, err = b.createAtx(ctx)
```

Comment:
why not handle poet errors here?

Reply:
if we will retry here we will miss the code block on top that discards stale challenge and builds a new one